TransformationSystem/scripts/dirac-production-runjoblocal.py

```diff
@@ -0,0 +1,166 @@
+  """
+  Creates the environment for running the job and returns
+  the path for the other functions.
+  
+  """
+
+  
+  tempdir = str(vo) + "job" + str(jobID) + "temp"
+
+  try:
+    os.mkdir(tempdir)
+    if not sys.exc_info()[1][0]:
+      S_OK("Temporary directory created.")    
+    pass
+  except:    
+    if sys.exc_info()[1][0] == 17:
+      S_OK("Temporary directory already exists.")
+    elif sys.exc_info()[1][0] == 30:
+      print sys.exc_info()[1], "Unable to create temporary directory"
+#      DError(errno.EROFS, "Unable to create temporary directory")
+
+    
+  basepath = os.getcwd()
+  return basepath + os.path.sep + tempdir + os.path.sep
+
+def __downloadJobDescriptionXML(jobID, basepath):
+  """
+  Downloads the jobDescription.xml file into the temporary directory
+  created.
+  
+  """
+  from DIRAC.Interfaces.API.Dirac import Dirac
+  jdXML = Dirac()
+  jdXML.getInputSandbox(jobID, basepath)
+
+def __modifyJobDescription(jobID, basepath, downloadinputdata):
+  """
+  Modifies the jobDescription.xml to, instead of DownloadInputData, it 
+  uses InputDataByProtocol
+  
+  """
+  if not downloadinputdata:
+    from xml.etree import ElementTree as et
+    archive = et.parse(basepath + "InputSandbox" + str(jobID) + os.path.sep + "jobDescription.xml")
+    for element in archive.getiterator():
+      if element.text == "DIRAC.WorkloadManagementSystem.Client.DownloadInputData":
+        element.text = "DIRAC.WorkloadManagementSystem.Client.InputDataByProtocol"
+        archive.write(basepath + "InputSandbox" + str(jobID) + os.path.sep + "jobDescription.xml")
+        S_OK("Job parameter changed from DownloadInputData to InputDataByProtocol.")
+
+  
+def __downloadPilotScripts(basepath, diracpath):
+  """
+  Downloads the scripts necessary to configure the pilot
+  
+  """
+  shutil.copyfile(str(diracpath) + os.path.sep + "WorkloadManagementSystem/PilotAgent/dirac-pilot.py"   , basepath + "dirac-pilot.py")
+  shutil.copyfile(str(diracpath) + os.path.sep + "WorkloadManagementSystem/PilotAgent/pilotCommands.py" , basepath + "pilotCommands.py")
+  shutil.copyfile(str(diracpath) + os.path.sep + "WorkloadManagementSystem/PilotAgent/pilotTools.py"    , basepath + "pilotTools.py")
+      
+def __configurePilot(basepath, vo):
+  """
+  Configures the pilot.
+  This method was created specifically for LHCb pilots, more info
+  about othe VOs is needed to make it more general.
+  """
+
+  from DIRAC.ConfigurationSystem.Client.Helpers.CSGlobals    import getVO, getSetup
+  from DIRAC.ConfigurationSystem.Client.ConfigurationData    import gConfigurationData
+  
+
+  
+  vo = getVO()
+  currentSetup = getSetup()
+  masterCS = gConfigurationData.getMasterServer()
+
+  os.system("python " + basepath + "dirac-pilot.py -S %s -l %s -C %s -N ce.debug.ch -Q default -n DIRAC.JobDebugger.ch -dd" %(currentSetup, vo, masterCS))
+  
+  dir = str(os.getcwd()) + os.path.sep
+  os.rename(dir + '.dirac.cfg', dir + '.dirac.cfg.old')
```

Comment:
2 things:
- .dirac.cfg may or may not exist
- .dirac.cfg if it exists is in ~/

so:
```
dir = os.path.expanduser('~') + os.path.sep
try:
  os.rename(dir + '.dirac.cfg', dir + '.dirac.cfg.old')
except OSError:
  pass
```